src/dialogs/more-info/controls/more-info-media_player.ts

```diff
@@ -0,0 +1,382 @@
+
+import "../../../components/ha-paper-dropdown-menu";
+import "../../../components/ha-icon-button";
+import "../../../components/ha-slider";
+import "../../../components/ha-icon";
+
+@customElement("more-info-media_player")
+class MoreInfoMediaPlayer extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ attribute: false }) public stateObj?: MediaEntity;
+
+  @query("#ttsInput") private _ttsInput?: HTMLInputElement;
+
+  protected render(): TemplateResult {
+    if (!this.stateObj) {
+      return html``;
+    }
+
+    const controls = this._getControls();
+    const stateObj = this.stateObj;
+
+    return html`
+      ${!controls
+        ? ""
+        : html`
+            <div class="controls">
+              ${controls!.map(
+                (control) => html`
+                  <ha-icon-button
+                    action=${control.action}
+                    .icon=${control.icon}
+                    @click=${this._handleClick}
+                  ></ha-icon-button>
+                `
+              )}
+            </div>
+          `}
+      ${(supportsFeature(stateObj, SUPPORT_VOLUME_SET) ||
+        supportsFeature(stateObj, SUPPORT_VOLUME_BUTTONS)) &&
+      ![UNAVAILABLE, UNKNOWN, "off"].includes(stateObj.state)
+        ? html`
+            <div class="volume">
+              ${supportsFeature(stateObj, SUPPORT_VOLUME_MUTE)
+                ? html`
+                    <ha-icon-button
+                      .icon=${stateObj.attributes.is_volume_muted
+                        ? "hass:volume-off"
+                        : "hass:volume-high"}
+                      @click=${this._toggleMute}
+                    ></ha-icon-button>
+                  `
+                : ""}
+              ${supportsFeature(stateObj, SUPPORT_VOLUME_SET)
+                ? html`
+                    <ha-slider
+                      id="input"
+                      pin
+                      ignore-bar-touch
+                      .dir=${computeRTLDirection(this.hass!)}
+                      .value=${Number(stateObj.attributes.volume_level) * 100}
+                      @change=${this._selectedValueChanged}
+                    ></ha-slider>
+                  `
+                : supportsFeature(stateObj, SUPPORT_VOLUME_BUTTONS)
+                ? html`
+                    <ha-icon-button
+                      action="volume_down"
+                      icon="hass:volume-minus"
+                      @click=${this._handleClick}
+                    ></ha-icon-button>
+                    <ha-icon-button
+                      action="volume_up"
+                      icon="hass:volume-plus"
+                      @click=${this._handleClick}
+                    ></ha-icon-button>
+                  `
+                : ""}
+            </div>
+          `
+        : ""}
+      ${stateObj.state === "off" ||
+      !supportsFeature(stateObj, SUPPORT_SELECT_SOURCE) ||
+      !stateObj.attributes.source_list
```

Comment:
Can we turn this around?
```suggestion
      ${stateObj.state !== "off" &&
      supportsFeature(stateObj, SUPPORT_SELECT_SOURCE) &&
      stateObj.attributes.source_list
```

Reply:
And should we check if `source_list` has a length?